app/tabs/_layout.tsx

```diff
@@ -0,0 +1,82 @@
+import { Ionicons, MaterialCommunityIcons } from "@expo/vector-icons";
+import { Platform, View } from "react-native";
+import Header from "../header";
+import React from "react";
+import Tabs from "../../components/Tabs";
+
+const AppLayout = () => {
+  const [headerVisibility, setHeaderVisibility] = React.useState(true);
+
+  return (
+    <>
+      <View style={{ marginTop: Platform.OS === "web" ? 0 : 30 }} />
+      {headerVisibility && (
+        <Header
+          title="onlyface"
+          messenger
+          search
+          plus
+          titleColor="#1a74e4"
+          style={{ backgroundColor: "white" }}
+        />
+      )}
+      <Tabs
+        screenListeners={({ route }) => ({
+          swipeEnd: () =>
+            setHeaderVisibility(route.name === "feed" || Platform.OS === "web"),
+        })}
+        backBehavior="history"
+        initialRouteName="feed"
+        screenOptions={({ route }) => ({
+          tabBarShowLabel: false,
+          lazy: true,
+          tabBarPressColor: "#e0e0e4",
+          tabBarIcon: ({ focused }) => {
+            let iconName;

```

Comment:
create a separate function component to render this, do not render inside the function of an element.